code/game/objects/items/radio/radio.dm

```diff
@@ -403,12 +405,26 @@
 					  filter_type, signal.data["compression"], list(position.z), connection.frequency,verb,language)
 
 
-/obj/item/radio/hear_talk(mob/M, msg, verb = "says", datum/language/language)
-	if(!language)
-		language = GLOB.language_datum_instances[get_default_language()]
-	if (broadcasting)
-		if(get_dist(src, M) <= canhear_range)
-			talk_into(M, msg, null, verb, language)
+/obj/item/radio/Hear(message, atom/movable/speaker, message_language, raw_message, radio_freq, list/spans, message_mode)
+	. = ..()
+
+	if(radio_freq || !broadcasting || get_dist(src, speaker) > canhear_range)
+		return
+
+	if(message_mode == MODE_WHISPER || message_mode == MODE_WHISPER_CRIT)
```

Comment:
Could use a simple `switch(message_mode)`, but don't bother if you think it's too much effort.

Reply:
Yeah this will get replaced in the next PR anyway